File: 0561. Array Partition/0561.java
// time complexity: O(nlogn)
// space complexity: O(1)
class Solution {
    public int arrayPairSum(int[] nums) {
        Arrays.sort(nums); // O(nlogn)
        int max = 0;
        // iterate on every pair
        for(int i=0; i<nums.length-1; i+=2){
            max += Math.min(nums[i], nums[i+1]);
        }
        return max;
    }
}

File: 0561. Array Partition/0561.py
# time complexity: O(nlogn)
# space complexity: O(1)
class Solution(object):
    def arrayPairSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums.sort() # O(nlogn)

        maximum = 0
        # sum up the even indices
        for i in range(0,len(nums),2):
            maximum += nums[i]
        
        return maximum